rules/consistent-indexof-check.js

```diff
@@ -0,0 +1,130 @@
+'use strict';
+const evaluateLiteralUnaryExpression = require('./utils/evaluate-literal-unaryexpression.js');
+const {isMethodCall} = require('./ast/index.js');
+
+const MESSAGE_ID = 'consistent-indexof-check';
+const messages = {
+	[MESSAGE_ID]: 'Prefer `{{replacement}}` over `{{value}}`.',
+};
+
+const comparisonMap = {
+	'<': '>',
+	'<=': '>=',
+	'>': '<',
+	'>=': '<=',
+};
+
+/**
+Determine the appropriate replacement based on the operator and value.
+
+@param {string} operator
+@param {number} value
+@param {string} variableName
+@returns {string | undefined}
+*/
+function getReplacement(operator, value, variableName) {
+	if ((operator === '<' && value <= 0) || (operator === '<=' && value <= -1)) {
+		return `${variableName} === -1`;
+	}
+
+	if ((operator === '>' && value === -1) || (operator === '>=' && value === 0)) {
+		return `${variableName} !== -1`;
+	}
+}
+
+/**
+Check if the node is a number literal or a unary expression resolving to a number.
+
+@param {import('estree').Node} node
+@returns {node is import('estree').Literal}
+*/
+function isNumberLiteral(node) {
+	if (node.type === 'UnaryExpression' && ['-', '+', '~'].includes(node.operator)) {
+		return isNumberLiteral(node.argument);
+	}
+
+	return node.type === 'Literal' && typeof node.value === 'number';
+}
+
+/**
+Find and process references to a given identifier.
+
+@param {import('eslint').Rule.RuleContext} context
+@param {import('estree').Identifier} identifierNode
+@returns
+*/
+function findAndProcessReferences(context, identifierNode) {
+	const variable = context.sourceCode.getDeclaredVariables(identifierNode.parent).find(variable => variable.name === identifierNode.name);
+
+	if (!variable) {
+		return;
+	}
+
+	for (const reference of variable.references) {
+		if (reference.identifier === identifierNode) {
+			continue;
+		}
+
+		const {identifier} = reference;
+		/** @type {{parent: import('estree').Node}} */
+		const {parent} = identifier;
+
+		// Check if the identifier is used in a binary expression
+		if (parent.type === 'BinaryExpression' && [parent.left, parent.right].includes(identifier)) {
+			const [literal, operator] = [parent.left, parent.right].some(n => isNumberLiteral(n))
+				? (parent.left === identifier
+					? [evaluateLiteralUnaryExpression(parent.right), parent.operator] // Index === -1
+					: [evaluateLiteralUnaryExpression(parent.left), comparisonMap[parent.operator]]) // -1 === index
+				: [];
+
+			const replacement = getReplacement(
+				operator,
+				literal,
+				identifier.name,
+			);
+
+			if (!replacement) {
+				continue;
+			}
+
+			context.report({
```

Comment:
This `context.report()` can not replace with a return statement

Otherwise, It can not report multiple issues when it returns.